firmware/common/i2cdevmanager.cpp

```diff
@@ -343,3 +344,17 @@ msg_t I2CDevManager::timer_fn(void* arg) {
 }
 
 };  // namespace i2cdev
+
+#ifdef __cplusplus
```

Comment:
since this is a .cpp file, #ifdef __cplusplus will alsways result in TRUE

use
> extern "C" int oNofityI2cFromShell(uint8_t* buff, size_t len)

as signature